src/state/session/index.tsx

```diff
@@ -283,6 +287,7 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
       )
 
       __globalAgent = agent
+      await fetchingGates
```

Comment:
The pattern I'm using here is to kick it off as soon as we have the DID, but to await just before the state change.